From: Aron Xu <aron@debian.org>
Date: Sat, 4 Feb 2012 15:30:56 +0800
Subject: [PATCH] Fix bashisms in scripts.

This commit fixes bashisms in scripts that are targeted to user
systems, excluding developer-targeted own convenient scripts. This
means they are complying with POSIX standards now and should be
working with any POSIX compliant shell.
---
 data/script/fcitx-configtool           |    4 ++--
 src/ui/classic/fcitx-skin-installer.sh |   10 +++++-----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/data/script/fcitx-configtool b/data/script/fcitx-configtool
index 592ddb3..0b573ac 100755
--- a/data/script/fcitx-configtool
+++ b/data/script/fcitx-configtool
@@ -63,8 +63,8 @@ run_xdg()
 }
 
 if [ ! -n "$DISPLAY" ]; then
-    [ -n "$EDITOR" ] && which $EDITOR >& /dev/null && exec $EDITOR $HOME/.config/fcitx/config
-    [ -n "$VISUAL" ] && which $VISUAL >& /dev/null && exec $VISUAL $HOME/.config/fcitx/config
+    [ -n "$EDITOR" ] && which $EDITOR > /dev/null 2>&1 && exec $EDITOR $HOME/.config/fcitx/config
+    [ -n "$VISUAL" ] && which $VISUAL > /dev/null 2>&1 && exec $VISUAL $HOME/.config/fcitx/config
     echo 'Please run it under X, or set $EDITOR or $VISUAL'
     exit 0
 fi
diff --git a/src/ui/classic/fcitx-skin-installer.sh b/src/ui/classic/fcitx-skin-installer.sh
index db3a8aa..bae110b 100644
--- a/src/ui/classic/fcitx-skin-installer.sh
+++ b/src/ui/classic/fcitx-skin-installer.sh
@@ -13,28 +13,28 @@ else
     MESSAGE_COMMON="echo"
 fi
 
-function zenity_error()
+zenity_error()
 {
     zenity --error --text="$1"
 }
 
-function zenity_info()
+zenity_info()
 {
     zenity --info --text="$1"
 }
 
-function usage()
+usage()
 {
     echo "Usage: fcitx-skin-installer [skin file]"
     exit 1
 }
 
-function message()
+message()
 {
     $MESSAGE_COMMON "$1"
 }
 
-function error()
+error()
 {
     $MESSAGE_ERROR "$1"
 }
-- 
